CRM/Activity/BAO/Activity.php

```diff
@@ -217,15 +217,6 @@ public static function deleteActivity(&$params, $moveToTrash = FALSE) {
       self::logActivityAction($activity, $logMsg);
     }
 
-    // delete the recently created Activity
```

Comment:
I just r-run tested this one & it worked (although the recent items block does require a screen refresh)